lib/auth/helpers.go

```diff
@@ -560,6 +560,35 @@ func (a *TestAuthServer) NewTestTLSServer() (*TestTLSServer, error) {
 	return srv, nil
 }
 
+// NewTestTLSServerWithConfig returns new test TLS server using the given
+// TestTLSServerConfig, overriding the auth server.
+func (a *TestAuthServer) NewTestTLSServerWithConfig(cfg TestTLSServerConfig) (*TestTLSServer, error) {
```

Comment:
Something about the naming feels off here. We already have
```
func NewTestTLSServer(cfg TestTLSServerConfig) (*TestTLSServer, error)
```

Which takes a config.

So now we'll have `NewTestTLSServer` and `NewTestTLSServerWithConfig` and both take configs?

Reply:
We also have this:
```
func (a *TestAuthServer) NewTestTLSServer() (*TestTLSServer, error) 
```

I've removed the additional function I added and open-coded the logic in `NewTestServer()`. Is this better?